test/langtools/tools/jnativescan/TestJNativeScan.java

```diff
@@ -0,0 +1,215 @@
+                .stdoutShouldContain("java.lang.foreign.MemorySegment::reinterpret(long)MemorySegment")
+                .stdoutShouldContain("org.service")
+                .stdoutShouldContain("org.service.ServiceImpl")
+                .stdoutShouldContain("org.service.ServiceImpl::m()void is a native method declaration")
+                .stdoutShouldContain("org.service.ServiceImpl::doIt()void references restricted methods")
+                .stdoutShouldContain("java.lang.foreign.MemorySegment::reinterpret(long)MemorySegment");
+    }
+
+    @Test
+    public void testAllModulePath() {
+        assertSuccess(jnativescan("--module-path", MODULE_PATH, "--add-modules", "ALL-MODULE-PATH"))
+                .stderrShouldBeEmpty()
+                .stdoutShouldContain("org.singlejar")
+                .stdoutShouldContain("org.lib")
+                .stdoutShouldContain("org.service");
+    }
+
+    @Test
+    public void testClassPathAttribute() {
+        assertSuccess(jnativescan("--class-path", classPathApp.toString()))
+                .stderrShouldBeEmpty()
+                .stdoutShouldContain("ALL-UNNAMED")
+                .stdoutShouldContain("lib.Lib")
+                .stdoutShouldContain("lib.Lib::m()void is a native method declaration")
+                .stdoutShouldContain("lib.Lib::doIt()void references restricted methods")
+                .stdoutShouldContain("java.lang.foreign.MemorySegment::reinterpret(long)MemorySegment");
+    }
+
+    @Test
+    public void testInvalidRelease() {
+        assertFailure(jnativescan("--module-path", MODULE_PATH, "--add-modules", "ALL-MODULE-PATH", "--release", "asdf"))
+                .stderrShouldContain("Invalid release");
+    }
+
+    @Test
+    public void testReleaseNotSupported() {
+        assertFailure(jnativescan("--module-path", MODULE_PATH, "--add-modules", "ALL-MODULE-PATH", "--release", "9999999"))
+                .stderrShouldContain("Release: 9999999 not supported");
+    }
+
+    @Test
+    public void testFileDoesNotExist() {
+        assertFailure(jnativescan("--class-path", "non-existent.jar"))
+                .stderrShouldContain("File does not exist, or does not appear to be a regular jar file");
+    }
+
+    @Test
+    public void testModuleNotAJarFile() {
+        String modulePath = moduleRoot("org.myapp").toString() + File.pathSeparator + orgLib.toString();
+        assertFailure(jnativescan("--module-path", modulePath,
+                        "--add-modules", "ALL-MODULE-PATH"))
+                .stderrShouldContain("File does not exist, or does not appear to be a regular jar file");
+    }
+
+    @Test
+    public void testPrintNativeAccess() {
+        assertSuccess(jnativescan("--module-path", MODULE_PATH,
+                                  "-add-modules", "org.singlejar,org.myapp",
+                                  "--print-native-access"))
+                .stdoutShouldContain("org.singlejar")
```

Comment:
It is a small thing, bu was there a consideration for a stronger assert on the output, checking that the output is precisely something like `--enable-native-access org.lib,org.service,org.singlejar`? Would require that the output is stable, which may be tricky, but also not a bad property. Just an idea for consideration.

Reply:
Good point. The result should be 'clean' and directly forward-able to `--enable-native-access`. (So in this case it should be exactly `org.singlejar`)

Reply:
I'm fixing this test, and also making the output sorted. That is required for the output stability, but I also thought that would be a nice property in case a user wants to diff the tool's outputs.